python/lsst/ts/atspectrograph/atspec_csc.py

```diff
@@ -635,61 +636,97 @@ async def configure(self, config):
+            filters_str['filter_central_wavelength'] += str(config.filters['filter_central_wavelength'][i])
+            filters_str['filter_focus_offset'] += str(config.filters['filter_focus_offset'][i])
+            # need to add comma, except for the last value
+            if i < len(self.model.filter_to_enum_mapping_dict) - 1:
+                # loop over keys to add a comma for each
+                for key in filters_str:
+                    filters_str[key] += ','
+
+        gratings_str = {'grating_name': '', 'grating_focus_offset': ''}
+        for i, f in enumerate(self.model.gratings_to_enum_mapping_dict):
+            gratings_str['grating_name'] += str(f)
+            gratings_str['grating_focus_offset'] += str(config.gratings['grating_focus_offset'][i])
+            # need to add comma, except for the last value
+            if i < len(self.model.gratings_to_enum_mapping_dict) - 1:
+                # loop over keys to add a comma for each, do not add a space after the comma!
+                for key in gratings_str:
+                    gratings_str[key] += ','
 
         if hasattr(self, "evt_settingsAppliedValues"):
             self.evt_settingsAppliedValues.set_put(host=self.model.host,
                                                    port=self.model.port,
                                                    linearStageMinPos=self.model.min_pos,
                                                    linearStageMaxPos=self.model.max_pos,
                                                    linearStageSpeed=0.,
-                                                   filterNames=filters_str,
-                                                   gratingNames=gratings_str,
+                                                   filterNames=filters_str['filter_name'],
+                                                   filterCentralWavelengths=filters_str['filter_central_wavelength'],
+                                                   filterFocusOffsets=filters_str['filter_focus_offset'],
+                                                   gratingNames=gratings_str['grating_name'],
+                                                   gratingFocusOffsets=gratings_str['grating_focus_offset'],
                                                    instrumentPort=config.instrument_port)
+
         elif hasattr(self, "evt_settingsApplied"):
             self.evt_settingsApplied.set_put(host=self.model.host,
                                              port=self.model.port,
                                              linearStageMinPos=self.model.min_pos,
                                              linearStageMaxPos=self.model.max_pos,
                                              linearStageSpeed=0.,
-                                             filterNames=filters_str,
-                                             gratingNames=gratings_str,
+                                             filterNames=filters_str['filter_name'],
```

Comment:
So, this will never be supported by the old `evt_settingsApplied`... 

I added this to allow backward compatibility with older versions of xml. Now that we are way ahead we could break this backward compatibility. I would propose remove this if/else statement altogether and just leave the `self.evt_settingsAppliedValues.set_put(...)` line